theano/tensor/nlinalg.py

```diff
@@ -625,9 +625,12 @@ def perform(self, node, inputs, outputs):
         (x,) = inputs
         (w, u, v) = outputs
         assert x.ndim == 2, "The input of svd function should be a matrix."
-        w[0], u[0], v[0] = self._numop(x,
-                                       self.full_matrices,
-                                       self.compute_uv)
+        if self.compute_uv:
+            w[0], u[0], v[0] = self._numop(x,
+                                           self.full_matrices,
+                                           self.compute_uv)
+        else:
+            u[0] = self._numop(x, self.full_matrices, self.compute_uv)
```

Comment:
I don't know which version of numpy is installed, but I think your fix is good and don't depend of numpy version. What trouble me is that why tests didn't catch that crash? Do you have a Theano test that crash without that fix? I suppose not, so this would mean this case isn't tested.